### DISCUSSION TITLE
Can kyuubi submit Scala, Python, and SQL code snippets

### DISCUSSION THREAD

**Discussion by smileyboy2019** [apache/kyuubi#6264] [Q&A]
Is it possible to integrate Livy and submit code snippets through a session? In addition to SQL queries, it is also possible to submit Java, Python, shell and other related code, supporting parameter passing and context

**Comment by pan3793** [comment]
Kyuubi has similar functionalities but not exactly same as Livy, check
```
org.apache.kyuubi.jdbc.hive.KyuubiStatement#executeScala
org.apache.kyuubi.jdbc.hive.KyuubiStatement#executePython
```
